pkg/sloop/queries/timerange.go

```diff
@@ -36,7 +36,22 @@ const minLookback = 1 * time.Minute
 // TODO: If wall clock is in the middle of the newest partition min-max time we can use it
 func computeTimeRange(params url.Values, tables typed.Tables, maxLookBack time.Duration) (time.Time, time.Time, error) {
 	endOfTime := getEndOfTime(tables)
-	return computeTimeRangeInternal(params, endOfTime, maxLookBack)
+	actualEndTime := computeActualEndTime(params.Get(EndTimeParam), endOfTime)
```

Comment:
`computeTimeRangeInternal` already takes into account `params.Get(EndTimeParam)`, not sure what we are trying to do here.

Reply:
@nurland endOfTime is `now`, params.Get(EndTimeParam) is user defined end time, we want to compare which one is earlier, and take the earlier one as the actual end time. Please see the computeActualEndTime function below, which explains how to determine the actualEndTime

Reply:
@mengyaoyang11 endOfTime is the latest available partition in badgerDb.
If you look at he following lines, we already account for user defined end time and we only override it when computedEnd (based on user input) is beyond endOfTime(latest partition) 
https://github.com/salesforce/sloop/pull/146/files#diff-3f5eeba9668570e88b8263b0d459cc5836ed590fde2bacfcd85a25b30ec3973eR88
https://github.com/salesforce/sloop/pull/146/files#diff-3f5eeba9668570e88b8263b0d459cc5836ed590fde2bacfcd85a25b30ec3973eR95

Reply:
That been said - we only do that when we pass startTime and endTime, when we pass LookbackInterval we set computedEnd to the endOfTime. I suggest we make change in https://github.com/salesforce/sloop/pull/146/files#diff-3f5eeba9668570e88b8263b0d459cc5836ed590fde2bacfcd85a25b30ec3973eR81 and keep endOfTime intact

Reply:
yes agreed this would be much cleaner. So in summary it would be:
1. Keep endOftime intact
2. Make change in computeTimeRangeInternal function where the endTime is set to endOfTime

Reply:
@nurland The links in you review directs me to the diff files of my PR, instead of the code lines you referred to, could you please repost it so I known which lines of code you are talking about? Thanks!